src/components/Modules/UseSessionsBatch.tsx

```diff
@@ -1,192 +1,181 @@
+  showSuccessMessage,
+  showErrorMessage,
+} from "../../utils";
+import {
+  DEFAULT_BATCHED_SESSION_ROUTER_MODULE,
+  DEFAULT_SESSION_KEY_MANAGER_MODULE,
+} from "@biconomy/modules";
+import { polygonMumbai } from "viem/chains";
+
+const ERC20RouterTransfer: React.FC = () => {
+  const classes = useStyles();
+  const { address } = useAccount();
+  const signer = useEthersSigner();
+  const { smartAccount, scwAddress } = useSmartAccountContext();
+  const [loading, setLoading] = useState(false);
+
+  const erc20Transfer = async () => {
+    if (!scwAddress || !smartAccount || !address) {
+      showErrorMessage("Please connect wallet first");
+      return;
+    }
+    try {
+      setLoading(true);
+      let biconomySmartAccount = smartAccount;
+      const managerModuleAddr = DEFAULT_SESSION_KEY_MANAGER_MODULE;
+      const erc20ModuleAddr = "0x000000D50C68705bd6897B2d17c7de32FB519fDA";
+      const routerModuleAddr = DEFAULT_BATCHED_SESSION_ROUTER_MODULE;
+      const mockSessionModuleAddr =
+        "0x7Ba4a7338D7A90dfA465cF975Cc6691812C3772E";
+
+      // get session key from local storage
+      const sessionKeyPrivKey = window.localStorage.getItem("sessionPKey");
+
+      if (!sessionKeyPrivKey) {
+        showErrorMessage("Session key not found");
+        return;
+      }
+
+      const provider = new ethers.providers.JsonRpcProvider(polygonMumbai.rpcUrls.public.http[0]);
+      const sessionSigner = new ethers.Wallet(sessionKeyPrivKey, provider);
+      console.log("sessionSigner", sessionSigner);
+
+      // generate sessionModule
+      const sessionModule = await SessionKeyManagerModule.create({
+        moduleAddress: managerModuleAddr,
+        smartAccountAddress: scwAddress,
+      });
+      const sessionRouterModule = await BatchedSessionRouterModule.create({
+        moduleAddress: routerModuleAddr,
+        sessionKeyManagerModule: sessionModule,
+        smartAccountAddress: scwAddress,
+      });
+
+      // set active module to sessionRouterModule
+      biconomySmartAccount =
+        biconomySmartAccount.setActiveValidationModule(sessionRouterModule);
+
+      // er20 transfer data generation
+      const tokenContract = new ethers.Contract(
+        config.usdc.address,
+        config.usdc.abi,
+        signer
+      );
+      let decimals = 18;
+      try {
+        decimals = await tokenContract.decimals();
+      } catch (error) {
+        throw new Error("invalid token address supplied");
+      }
+      const amountGwei = ethers.utils.parseUnits("0.1".toString(), decimals); // MAKE SURE SCW HAS ENOUGH USDC, otherwise user op will fail
+      const data = (
+        await tokenContract.populateTransaction.transfer(
+          "0x42138576848E839827585A3539305774D36B9602", // receiver address
+          amountGwei
+        )
+      ).data;
+      const data2 = (
+        await tokenContract.populateTransaction.transfer(
+          "0x5a86A87b3ea8080Ff0B99820159755a4422050e6", // receiver address 2
```

Comment:
can add comments like this data corresponds to session using mock svm so anything can be done. can be different receiver etc